### ISSUE TITLE
MacOS, The developer cannot be verified

### ISSUE THREAD

**Issue by Alex2782** [godotengine/webrtc-native#124]
### Godot version

v4.1.2.stable.official [399c9dc39]

### Plugin version

v1.0.3

### System information

macOS 13.6.0 - Vulkan (Forward+) - integrated Apple M1 - Apple M1 (8 Threads)

### Issue description

System message in German language. `The developer cannot be verified`
Godot tries to load the lib several times, cancel not possible, MessageBox comes again and again. The application must be terminated in the Task Manager. 

https://github.com/godotengine/webrtc-native/releases
`godot-extension-4.1-webrtc.zip`

![image](https://github.com/godotengine/webrtc-native/assets/41921395/a8fdb048-5c4a-49d3-9e62-f1cc673203fe)

### Steps to reproduce

-

**Comment by aidan-j-rhoden** [comment]
Open Settings, then Privacy & Security.  Click the button for that app that says, "Open Anyway."  This button will be visible for about an hour after you attempted to open it.
[support.apple.com/guide/mac-help/open-a-macapp-from-an-unidentified-developer](https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/13.0/mac/13.0#:~:text=Note%3A%20You%20can%20also%20grant%20an%20exception%20for%20a%20blocked%20app%20by%20clicking%20the%20Open%20Anyway%20button%20in%20Privacy%20%26%20Security%20settings.%20This%20button%20is%20available%20for%20about%20an%20hour%20after%20you%20try%20to%20open%20the%20app.)

**Comment by Alex2782** [comment]
Probably it is better to compile the webrtc-native sources yourself and run [code-signing-and-notarization](https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_macos.html#code-signing-and-notarization)
Otherwise all users would have to do that (customize Privacy & Security)? That's out of the question, I think.

**Comment by aidan-j-rhoden** [comment]
Yes, that would definitely be better practice, but this is a quick workaround until you can get a Apple Developer ID, which, I assume, is a hassle.
With any exported projects from Godot to Mac, your end users would experience the same issue, unless you were able to code-sign and notarize the builds, which still requires an Apple Dev ID.
This might be helpful to you: [developer.apple.com/help/account/create-certificates/create-developer-id-certificates/](https://developer.apple.com/help/account/create-certificates/create-developer-id-certificates/)

**Comment by Ughuuu** [comment]
So I was wondering what's the status on this? @Faless do you currently have a mac developer account?

**Comment by Ughuuu** [comment]
Or whoever is in charge of this addon I mean.

**Comment by Faless** [comment]
> @Faless do you currently have a mac developer account?

I don't have a mac developer account so I can't sign it. We might want to sign it with the same identity that signs Godot, but I'm not sure (CC @akien-mga @hpvb )

**Comment by Ughuuu** [comment]
Also i try to build on ios and its failing at signing.(using Xcode localy)
It appears xcode is generating a plist file that is wrong or something.

**Comment by Ughuuu** [comment]
The exact error I am getting when trying to build for iOS is:

```
This bundle is invalid. The bundle at path Payload/polyglot.app/Frameworks/libwebrtc_native.ios.template_debug.arm64.framework has an invalid CFBundleIdentifier 'com.gdextension.framework.libwebrtc_native.ios.template_debug.arm64' There are invalid characters(characters that are not dots, hyphen and alphanumerics) that have been replaced with their code point 'com.gdextension.framework.libwebrtc\\u005fnative.ios.template\\u005fdebug.arm64' CFBundleIdentifier must be present, must contain only alphanumerics, dots, hyphens and must not end with a dot
```
and
```
Invalid Bundle. The bundle polyglot.app/Frameworks/libwebrtc_native.ios.template_debug.arm64.framework does not support the minimum OS Version specified in the Info.plist.
```

It only happens if I export the build to app store, if I run locally it runs fine.

**Comment by Ughuuu** [comment]
Aparently solution is to have a .plist file near the .dylib and have them instead of .dylib to be a .framework that contains a .plist and a .dylib. (for the iOS problem).
For the mac problem, we could use automatic signing action, eg similar to this I wrote: https://github.com/Ughuuu/godot-cpp-template/blob/add-more-stuff/options-to-build/.github/actions/sign/action.yml
If it gets approved into godot-cpp-template we could reuse it from there.

**Comment by Ughuuu** [comment]
I would want to change this but the build system is so overly complicated, it's not just calling SConstruct but has a lot of scripts here and there. If you need an example on how the Info.plist file should look like, there is one either in godot-cpp-template or in godot-jolt.
